lib/srv/discovery/discovery.go

```diff
@@ -1466,3 +1472,15 @@ func splitMatchers[T types.Matcher](matchers []T, matcherTypeCheck func(string)
 	}
 	return
 }
+
+func (s *Server) updatesEmptyDiscoveryGroup(getter func() (types.ResourceWithLabels, error)) bool {
```

Comment:
Is this function appropriately named? I don't see anything that gets updated, but I may be misunderstanding what this code does.

Maybe `isDiscoveryGroupEmpty` instead?

Reply:
this function checks if we are updating a resource with an empty discovery group. It requires going to the DB and pulling the resource version there because if it's empty we allow the update.

I tried to avoid `isDiscoveryGroupEmpty` because it gives the impression that it checks for the current version (discovery group not empty) while in reality, it checks the discovery group of the existing resource